sklearn/preprocessing/_discretization.py

```diff
@@ -116,8 +121,10 @@ class KBinsDiscretizer(BaseEstimator, TransformerMixin):
         ``1`` based on a parameter ``threshold``.
     """
 
-    def __init__(self, n_bins=5, encode='onehot', strategy='quantile'):
+    def __init__(self, n_bins=5, subsample=1e5, encode='onehot',
```

Comment:
Please put the new parameter at the end of the list (just before the missing `random_state` parameter). So as to minimize the likelihood of break users' code that unsafely pass constructor params positionally.

Also we cannot unfortunately change the default behavior of estimators without going through a deprecation warning cycle: I guess in this case we will need to set the default value of the  `subsample` constructor param to `subsample='warn'` to issue a `FutureWarning` that says that in the future  (0.22) `subsample=1e5` will be used by default and ask the user to pass `subsample` explicitly to silence this warning in the mean time.

Reply:
Done